src/utils/loadUmdBundle.ts

```diff
@@ -1,31 +1,86 @@
 import "@ungap/global-this";
 
-const umdBundlesPromiseCacheMap = new Map<string, Promise<unknown>>();
+import { retryFetch } from "./retryFetch";
+
+export interface LoadUmdBundleBackoff {
+  maxRetries: number;
+  delay: number;
+}
+
+interface UmdBundleCacheItem {
+  promise: Promise<unknown>;
+  timestamp: number;
+}
+
+const umdBundlesPromiseCacheMap = new Map<string, UmdBundleCacheItem>();
+
+export interface LoadUmdBundleOptions {
+  ttlInMs?: number;
+  backoff?: LoadUmdBundleBackoff;
+}
+
+const isCacheItemValid = ({
+  ttlInMs,
+  timestamp
+}: {
+  ttlInMs?: number;
+  timestamp: number;
+}) => !ttlInMs || Date.now() - timestamp < ttlInMs;
```

Comment:
You probably want to use `ttlInMs == null` instead, if I pass `ttlInMs` to 0 I'd expect it to disable the cache completely, but with `!ttlInMs` it would still use the cache 👍